packages/core/src/builder.ts

```diff
@@ -611,4 +617,46 @@ export default class SchemaBuilder<Types extends SchemaTypes> {
       ? processedSchema
       : lexicographicSortSchema(processedSchema);
   }
+
+  /**
+   * Helper for allowing plugins to fulfill the return of the `next` resolver, without paying the cost of the
+   * Promise if not required.
+   */
+  completeValue<T, R>(
+    valOrPromise: PromiseLike<T> | T,
+    onSuccess: (completedVal: T) => R,
+  ): R | Awaited<R> | Promise<Awaited<R>> | Promise<Awaited<R>>;
+
+  /**
+   * Helper for allowing plugins to fulfill the return of the `next` resolver, without paying the cost of the
+   * Promise if not required.
+   */
+  completeValue<T, R, E>(
+    valOrPromise: PromiseLike<T> | T,
+    onSuccess: (completedVal: T) => R,
+    onError: (err: unknown) => E,
+  ): R | Awaited<R> | Promise<Awaited<R>> | Promise<Awaited<R>>;
+
+  /**
+   * Helper for allowing plugins to fulfill the return of the `next` resolver, without paying the cost of the
+   * Promise if not required.
+   */
+  completeValue<T, R>(
```

Comment:
Any reason this needs to be on the builder rather than exported along with other utils?

Reply:
Yeah was just changing this to be a util fn